ynput/ayon-core
# Add simple Create Project Structure launcher action

Imp for [AY-5076](https://app.clickup.com/t/6658547/AY-5076)

## Changelog Description

Add simple Create Project Structure launcher action.

![image](https://github.com/ynput/ayon-core/assets/2439881/dd2b5d35-0040-4779-b1bc-240e5d8a67fa)

Which creates the folder structure as defined in project settings: `ayon+settings://core/project_folder_structure`

![image](https://github.com/ynput/ayon-core/assets/2439881/3031741d-a536-44ec-9a21-531fd2c31fe3)

## Additional info

It currently only shows when you're in a project in the launcher but have NO folder or task selected to avoid clutter in launcher.

The label unfortunately is a bit too long to be nicely displayed by default, see screenshot. **Any ideas for better icon and shorter label that doesn't get cut off?**

---

This feature request came up a few times:
- [Ynput Community Forum: How set up custom folder creation when using the launcher](https://community.ynput.io/t/how-to-set-up-custom-folder-creation-when-using-the-launcher-or-ftrack-actions/452/10?u=bigroy)
- [Ynput Community Forum: Default Project Hierarchy](https://community.ynput.io/t/default-project-hierarchy-made-of-folder-type-folder/871?u=bigroy) mentions the setting but seemed to describe a different feature.
- [Discord](https://discord.com/channels/517362899170230292/563751989075378201/1253392662036480142)

## Testing notes:

1. Update core addon
2. Run the launcher action when you enter a project.
3. Folders should now be generated as per settings.
